pkg/proxy/service.go

```diff
@@ -168,22 +164,25 @@ func (sct *ServiceChangeTracker) newBaseServiceInfo(port *v1.ServicePort, servic
 	}
 
 	// Obtain Load Balancer Ingress IPs
-	var ips []string
+	var allIncorrectIPs []string
 	for _, ing := range service.Status.LoadBalancer.Ingress {
-		ips = append(ips, ing.IP)
-	}
+		// []string{ing.IP} have a len of 1, so len(correctIPs) + len(incorrectIPs) == 1
```

Comment:
thank you for this explanation here!

Reply:
Indeed.  This loop is now sufficiently complicated that it should be broken out to a func and have a unit test.